Carbon.Domain.EntityFrameworkCore.Extensions/Repositories/TenantManagedEFRepositoryBase.cs

```diff
@@ -31,45 +31,44 @@ public TenantManagedEFRepositoryBase(DbContext ctx)
 
         public override async Task ConnectToSolution<T>(T relatedEntity)
         {
-            if (relatedEntity == null || relatedEntity.RelationalOwners == null || !relatedEntity.RelationalOwners.Any() || relatedEntity.RelationalOwners.Where(k => k == null).Any())
-                return;
-            bool relationsChanged = false;
-            
-            foreach (var ro in relatedEntity.RelationalOwners)
+			if (relatedEntity == null || relatedEntity.RelationalOwners.Where(k => k == null).Any())
```

Comment:
There should be a null check` relatedEntity == null || (relatedEntity.RelationalOwners != null && relatedEntity.RelationalOwners.Where(k => k == null))`